pcsx2/CDVD/CDVD.cpp

```diff
@@ -2247,21 +2249,115 @@ static void cdvdWrite16(u8 rt) // SCOMMAND
 						break;
 
 					case 0xFD: // _sceCdReadRenewalDate (1:6) BCD
-						SetSCMDResultSize(6);
-						cdvd.SCMDResult[0] = 0;
-						cdvd.SCMDResult[1] = 0x04; //year
-						cdvd.SCMDResult[2] = 0x12; //month
-						cdvd.SCMDResult[3] = 0x10; //day
-						cdvd.SCMDResult[4] = 0x01; //hour
-						cdvd.SCMDResult[5] = 0x30; //min
+						switch (temp_mechaver[1])
+						{
+							case 0x01:
+							case 0x02:
+							case 0x03:
+								// Mechacon version prior to v5 doesnt support SCMD 03,FD command
+								SetSCMDResultSize(1);
+								cdvd.SCMDResult[0] = 0x80;
+								break;
+
+							case 0x05:
+								SetSCMDResultSize(6);
+								switch (temp_mechaver[2])
+								{
+									case 0x00:
+									case 0x01:
+										memcpy(cdvd.SCMDResult, &MRenewalDate[0], 6);
+										break;
+									case 0x02:
+									case 0x03:
+										memcpy(cdvd.SCMDResult, &MRenewalDate[1], 6);
+										break;
+									case 0x04:
+									case 0x05:
+										memcpy(cdvd.SCMDResult, &MRenewalDate[2], 6);
+										break;
+									case 0x06:
+									case 0x07:
+										if (temp_mechaver[0] != 0x07)
+											memcpy(cdvd.SCMDResult, &MRenewalDate[3], 6);
+										else
+											memcpy(cdvd.SCMDResult, &MRenewalDate[4], 6);
+										break;
+									case 0x0A:
+									case 0x0B:
+										memcpy(cdvd.SCMDResult, &MRenewalDate[5], 6);
+										break;
+									case 0x0C:
+									case 0x0D:
+										memcpy(cdvd.SCMDResult, &MRenewalDate[6], 6);
+										break;
+									case 0x0E:
+									case 0x0F:
+										memcpy(cdvd.SCMDResult, &MRenewalDate[7], 6);
+										break;
+									default:
+										SetSCMDResultSize(1);
+										cdvd.SCMDResult[0] = 0x80;
+										break;
+								}
+								break;
```

Comment:
Could surely be shortened to just do 

`memcpy(cdvd.SCMDResult, &MRenewalDate[temp_mechaver[2] >> 1], 6);`

You'd just need a special case for 0x6-0x7 and checking it's less than 0x10 (though realistically, will it ever be higher?)

Reply:
wow thanks for C tip :)

Reply:
yes 0x6, 0x7 is a special mechacon (5.6) that was released in different variations for 2 regions: Mexico and other regions. As Mexico is region 07 it is enough to test region byte in mechacon response.